web/src/engine/dom-utils/src/stylesheets.ts

```diff
@@ -144,42 +162,55 @@ export class StylesheetManager {
         // with embedded ttf or woff.  svg mostly works so is a better initial
         // choice on the Android browser.
         if(svg != '') {
-          s0.push("url('"+svg+"') format('svg')");
+          source = "url('"+svg+"') format('svg')";
         }
 
         if(woff != '') {
-          s0.push("url('"+woff+"') format('woff')");
+          source = "url('"+woff+"') format('woff')";
         }
 
         if(ttf != '') {
-          s0.push("url('"+ttf+"') format('truetype')");
+          source = "url('"+ttf+"') format('truetype')";
         }
       } else {
         if(woff != '') {
-          s0.push("url('"+woff+"') format('woff')");
+          source = "url('"+woff+"') format('woff')";
         }
 
         if(ttf != '') {
-          s0.push("url('"+ttf+"') format('truetype')");
+          source = "url('"+ttf+"') format('truetype')";
         }
 
         if(svg != '') {
-          s0.push("url('"+svg+"') format('svg')");
+          source = "url('"+svg+"') format('svg')";
         }
       }
+    }
 
-      if(s0.length == 0) {
-        return null;
-      }
-
-      s += 'src:'+s0.join(',')+';';
+    if(!source) {
+      return null;
     }
 
+    s += 'src:'+source+';';
+
     s=s+'\n}\n';
 
     const sheet = createStyleSheet(s);
     fontStyleMap[fontKey] = sheet;
 
+    /* https://developer.mozilla.org/en-US/docs/Web/API/CSS_Font_Loading_API
+     * Compat:  Chrome 35... _just_ on the unupdated-Android 5.0 threshold.
+     *
+     * Note:  this could probably wholesale-replace the stylesheet!
+     * Would need: `document.fonts.add(fontFace)` - does not have to wait for the load() Promise.
+     *
+     * For now, we're using this solely to detect when the font has been succesfully loaded.
+     */
+    const fontFace = new FontFace(fd.family, source);
+    const loadPromise = fontFace.load();
+    this.fontPromises.push(loadPromise);
+    loadPromise.then(() => this.fontPromises = this.fontPromises.filter((entry) => entry != loadPromise));
```

Comment:
Wouldn't it be cleaner to just clear `fontPromises` after `allSettled()` on l.62 or so?

Reply:
There's technically a chance for another stylesheet to come in at a later point that would need a `Promise` not currently included.  It's best not to assume that there will only be one call to the `allLoadedPromise()` method - largely because, in #10507, documentation-keyboards do end up needing a separate call to the method.  (Though... I might actually be able to restructure that a bit.)

To me, this is the cleanest way - each Promise ensures that it's only included in the `allLoadedPromise` when it's actually still loading, bowing out as soon as it's completed.  If new ones rotate in and old ones rotate out in the meantime, that all comes out in the wash with the current approach.  That is, so long as I add `.catch` handling too - _that_ part is certainly an important "catch" here.

Reply:
Secondary, do `Promise.all()` (and cousins) handle the scenario of the array changing while they are waiting?